tests/jballerina-unit-test/src/test/resources/test-src/workers/worker_async_send_as_expression.bal

```diff
@@ -14,12 +14,51 @@
 // specific language governing permissions and limitations
 // under the License.
 
-function foo() {
+function testAsyncSendAsExpressionReturnType() {
     worker w1 {
         () a = 5 -> w2;
+        assertValueEquality(a, ());
+
+        var b = "Foo" -> w2;
+        assertValueEquality(b, ());
     }
 
     worker w2 {
-        int a = <- w1;
+        int x = <- w1;
+        assertValueEquality(5, x);
+
+        string y = <- w1;
+        assertValueEquality("Foo", y);
+    }
+
+    _ = wait {w1, w2};
+}
+
+function testAsyncSendAsExpressionWithPanic() {
+    worker w1 {
+        () a = 5 -> w2;
+        () b = 15 -> w2;
+        assertValueEquality(b, ());
+    }
+
+    worker w2 returns error? {
+        int x = <- w1;
+        if (x == 5) {
+            return error("This is an error");
+        }
+        x = <- w1;
+    }
+
+     map<error??> mapResult = wait {w1, w2};
+}
+
```

Comment:
Shall we add more tests to test the behavior of async send in all places where this is allowed to be used now? 
Check places where `action` and `action-or-expr` are allowed in the spec. ex:type casts, check expr, trap, lvexpr etc

Reply:
Added tests for this https://github.com/ballerina-platform/ballerina-lang/pull/38245/commits/00909f206139900d02b95963101209f7cb0f45e7